### PR TITLE
Basic UI

### PR DESCRIPTION
**Pull request by EddyShimwa**
### Create a `app.rb` file that will serve as your console app entry-point. It should have methods that do the following:
- List all books.
- List all people.
- Create a person (teacher or student, not a plain Person).
- Create a book.
- Create a rental.
- List all rentals for a given person id.
### In my `main.rb` I defined the entry point method to do the following:
- Present the user with a list of options to perform.
- Lets users choose an option.
- If needed, ask for parameters for the option.
- Have a way to quit the app.

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+require_relative 'app'
+
+def display_list
+  action_list = {
+    '1': 'List all books',
+    '2': 'List all people',
+    '3': 'Create a person',
```
Comment (Davidosky007) [inline main.rb:7]: * Kindly fix this error i get when i try to create a person 
image
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @EddyShimwa 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* No linter errors 👍 
* Well written PR description 👍 
* Good looking readme file 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by EddyShimwa** [comment]
Thanks @Davidosky007 for reviweing this PR

**Comment by EddyShimwa** [comment]
image

### I Updated it!

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+      colorize_outprint(35, 'Age: ')
+      age = gets.chomp
+
+      colorize_outprint(35, 'Has parent permission? [Y/N]: ')
+      permission = gets.chomp.downcase
+      case permission
+      when 'y' then parent_permission = true
+      when 'n' then parent_permission = false
+      end
+
```
Comment (ghost) [inline app.rb:63]: - [ ] Creating a teacher works fine but there is still an error when creating a student on your app. Kindly check it out.

image

image

**Review by frank1738 — CHANGES_REQUESTED** [review]
Hi @EddyShimwa , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights

✅ Correct use of Gitflow
✅ No linter error


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 2 above)
Comment (EddyShimwa) [reply app.rb:63]: Thanks @frank1738 for reviewing this PR

File: main.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    '4': 'Create a book',
+    '5': 'Create a rental',
+    '6': 'List all rentals for a given person id',
```
Comment (Mihndim2020) [inline main.rb:10]: - [ ] As it is a [requirement ](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_ui.md)for this project, "`List all rentals for a given person id.`", please ensure one is able to list the rentals for a particular `id`, that is not the case at the moment, please kindly fix that. What I get is an empty list.

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi @EddyShimwa, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark:  Clean and Organized code
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION

# Status: Requires Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+  def list_rentals_of_person_id()
+    print 'Person ID: '
+    id = gets.chomp.to_i
+
+    rentals = @rentals.filter { |rental| rental.person.id == id }
+    puts 'List of Rentals:'
+    rentals.each do |rental|
+      puts "Date: #{rental.date}, Book '#{rental.book.title}' by #{rental.book.author}"
+    end
```
Comment (TedLivist) [inline app.rb]: - [ ] As you can see from the image below when I try to list the rentals using an ID I get no response despite 'creating a rental'. If you look at line 147 you are trying to loop over the `@rentals` array defined earlier in the file, but the problem is that in the `create_rental` method, on line 134, you are not adding the created rentals in the first place. Kindly fix that by pushing the rental to the array.

  ```
  @rentals.push(Rental.new(date, @books[book_option], @people[person_option]))
  ```

  uurir

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @EddyShimwa,

✔️ Persons can be created

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 4 above)
Comment (EddyShimwa) [reply app.rb]: Thanks @TedLivist !

**Review by ShadyShawkat — APPROVED** [review]
## Hi @EddyShimwa, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**